### ISSUE TITLE
[Adjust Rule] SIM904 false positive (accessing previous dict values)

### ISSUE THREAD

**Issue by Skylion007** [MartinThoma/flake8-simplify#100]
## Desired change

* **Rule(s)**: SIM986, SIM987
* **Adjustment**: In which way should it/they be adjusted?

## Explanation
* The SIM904 check currently does not check if previous values were accessed in keys. This means new keys cannot access old keys since the dictionary is not yet constructed.
* 
Why is the adjustment necessary / better?

## Example

This is an example where the mentioned rule(s) would currently be suboptimal:

```python
        perf = {"total_time": end_time - start_time}
        perf["frame_time"] = perf["total_time"] / total_frames
        perf["fps"] = 1.0 / perf["frame_time"]
        perf["time_per_step"] = time_per_step
        perf["avg_sim_step_time"] = total_sim_step_time / total_frames
```

or more simply:

```python
        perf = {"a": 1}
        perf["b"] = perf["a"] / 10
```

Yes, in this case, it could be written by creating lots of additional variables to replace the references to previous values of "total_time" and "frame_time", but in more complicated dictionaries it seems like this rule could become more troublesome,

**Comment by MartinThoma** [comment]
Good catch, I'll fix that tomorrow.

**Comment by MartinThoma** [comment]
Just in case you missed it : in order to still add new rules quickly and don't disturb users of flake8-simplify, I'll use SIM9 rules. If there are no more false positives for 6 months, I'll change the rule to something not beginning with "9". This way people can ignore the SIM9 rules in their CI and just try them once in a while.

**Comment by MartinThoma** [comment]
This looks very similar to #99 . I'll address both in the same PR/commit
